Sources/AST/AST.swift

```diff
@@ -10,6 +10,100 @@ public struct TopLevelModule: Equatable {
+
+    let identifier = Identifier(identifierToken: Token(kind: .identifier("get"+capitalisedFirst), sourceLocation: variable.identifier.sourceLocation))
+    guard let (parameters, expression, resultType) = getProperty(identifier: variable.identifier, variableType: variable.type, sourceLocation: variable.sourceLocation) else {
+      return nil
+    }
+    let body = [Statement.returnStatement(ReturnStatement(returnToken: Token(kind: .return, sourceLocation: variable.sourceLocation), expression: expression))]
+
+    return FunctionDeclaration(funcToken: dummyFuncToken, attributes: [], modifiers: [Token(kind: .public, sourceLocation: variable.sourceLocation)], identifier: identifier, parameters: parameters, closeBracketToken: dummyCloseToken, resultType: resultType, body: body, closeBraceToken: dummyCloseToken)
+  }
+
+  func synthesizeMutator(variable: VariableDeclaration) -> FunctionDeclaration? {
+    let dummyFuncToken = Token(kind: .func, sourceLocation: variable.sourceLocation)
+    let dummyCloseToken = Token(kind: .punctuation(.closeBrace), sourceLocation: variable.sourceLocation)
+
+    let capitalisedFirst = variable.identifier.name.prefix(1).uppercased() + variable.identifier.name.dropFirst()
+
+    let identifier = Identifier(identifierToken: Token(kind: .identifier("set"+capitalisedFirst), sourceLocation: variable.identifier.sourceLocation))
+    guard let (parameters, expression, resultType) = getProperty(identifier: variable.identifier, variableType: variable.type, sourceLocation: variable.sourceLocation) else {
+      return nil
+    }
+    let valueIdentifier = Identifier(identifierToken: Token(kind: .identifier("value"), sourceLocation: variable.sourceLocation))
+    let valueParameter = Parameter(identifier: valueIdentifier, type: resultType, implicitToken: nil)
+    let body = [Statement.expression(.binaryExpression(BinaryExpression(lhs: expression, op: Token(kind: .punctuation(.equal), sourceLocation: variable.sourceLocation), rhs: .identifier(valueIdentifier))))]
+
+    return FunctionDeclaration(funcToken: dummyFuncToken, attributes: [], modifiers: [Token(kind: .public, sourceLocation: variable.sourceLocation), Token(kind: .mutating, sourceLocation: variable.sourceLocation)], identifier: identifier, parameters: parameters + [valueParameter], closeBracketToken: dummyCloseToken, resultType: nil, body: body, closeBraceToken: dummyCloseToken)
+  }
+
+  func getProperty(identifier: Identifier, variableType: Type, sourceLocation: SourceLocation) -> ([Parameter], Expression, Type)? {
+    switch variableType.rawType {
+      case .basicType(_):
+        return ([], .identifier(identifier), variableType)
+      case .arrayType(let type), .fixedSizeArrayType(let type, _):
+        var identifiers = [Identifier]()
+        var currentType: Type.RawType = type
+        var currentName: String
+        while true {
+          currentName = String(UnicodeScalar(Int(UnicodeScalar("i")!.value) + identifiers.count)!)
```

Comment:
This is not a nice way to implement this feature (not that I have any other suggestions).

Reply:
I would agree! I'll have a look to see if I can find something nicer